### ISSUE TITLE
Build Failed issue in  Building 5.5.9 into /home/shippable/.phpenv/versions/5.5.9

### ISSUE THREAD

**Issue by aftabaslam** [Shippable/support#3788]
[Info]: Building 5.5.9 into /home/shippable/.phpenv/versions/5.5.9
[Downloading]: https://secure.php.net/distributions/php-5.5.9.tar.bz2
[Preparing]: /tmp/php-build/source/5.5.9
[Compiling]: /tmp/php-build/source/5.5.9
[xdebug]: Installing version 2.5.5
[xdebug]: Compiling xdebug in /tmp/php-build/source/xdebug-2.5.5
[xdebug]: Installing xdebug configuration in /home/shippable/.phpenv/versions/5.5.9/etc/conf.d/xdebug.ini
[xdebug]: Cleaning up.
[Info]: Enabling Opcache...
[Info]: Done
[Info]: The Log File is not empty, but the Build did not fail. Maybe just warnings got logged. You can review the log in /tmp/php-build.5.5.9.20170823114926.log
[Success]: Built 5.5.9 successfully.
/usr/local/bin/phpenv-install: line 76: phpenv: command not found

**Comment by scriptnull** [comment]
Hi @aftabaslam,

Could you provide us with the build link ? If you prefer to send it privately, please drop a mail to vishnu@shippable.com 

Thanks.

**Comment by aftabaslam** [comment]
https://app.shippable.com/github/1eq/1EQ/runs/5193/summary/console

404 | Shippable<https://app.shippable.com/github/1eq/1EQ/runs/5193/summary/console>
app.shippable.com
Make devOps systematic - a devOps automation platform




Aftab Aslam






Plot No. 156, Software Technology Park, Sector I-9/3, Islamabad

<http://www.stellatechnology.com/>http://stellatechnology.com<http://stellatechnology.com/> | c: (+92) 345 50 80 588  | aaslam@stellatechnology.com<mailto:aaslam@stellatechnology.com><mailto:salim@stellatechnology.com>


________________________________

**Comment by aftabaslam** [comment]
i have send you email

**Comment by scriptnull** [comment]
Let's continue the conversation over email. I have sent you some instructions over mail. Please see if that resolves your problem.

**Comment by khurram85** [comment]
Hi @scriptnull 

We have updated the file as suggested now its failing at 

```composer global require phing/phing:2.* 0s  
rbenv: composer: command not found

The `composer' command exists in these PHP versions:
  5.4
  5.5
  5.6
  7.0
```
However, in the same build we can see the composer command with sudo worked just fine. 
```sudo composer self-update 0s  
You are already using composer version 1.5.1 (stable channel).
```
Could you please let us know what we are missing.

**Comment by AunR** [comment]
Hi
 It is failing with PHP 5.6 as well. Please help, its urgent issue.

- Aun

**Comment by aftabaslam** [comment]
composer global require phing/phing:2.6.1 0s  
rbenv: composer: command not found

The `composer' command exists in these PHP versions:
  5.4
  5.5
  5.6
  7.0

**Comment by scriptnull** [comment]
As we discussed over mail and call, I have opened an internal issue for this. We will try to resolve this as soon as possible.

**Comment by ambarish2012** [comment]
@aftabaslam here is what I would recommend -

- Set the ami for your subscription to be v5.8.2
- reinit the node
- run your build

Your custom node is now running AMI v5.8.3 and the problem started occuring after this change. If you need to get into a call with us to debug this, please let me know.

**Comment by AunR** [comment]
@ambarish2012 

We followed what you suggested above and build is still failing with exactly same error.

**Comment by rageshkrishna** [comment]
@AunR Looking at [build 5236](https://app.shippable.com/github/1eq/1EQ/runs/5236/1/console), I see that you are now getting exactly the same PHP version as your [last successful build (5164)](https://app.shippable.com/github/1eq/1EQ/runs/5164/1/console). The build environment is now identical in both cases, as judged from the output of `php -v` and the `USER` and `HOME` env vars.

However, I do see the following errors in the output of `phing installBuildDeps`:

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for aws/aws-sdk-php 2.8.31 -> satisfiable by aws/aws-sdk-php[2.8.31].
    - aws/aws-sdk-php 2.8.31 requires guzzlehttp/guzzle ~3.7 -> no matching package found.
  Problem 2
    - aws/aws-sdk-php 2.8.31 requires guzzlehttp/guzzle ~3.7 -> no matching package found.
    - league/flysystem-aws-s3-v2 1.0.3 requires aws/aws-sdk-php ~2.7 -> satisfiable by aws/aws-sdk-php[2.8.31].
    - Installation request for league/flysystem-aws-s3-v2 1.0.3 -> satisfiable by league/flysystem-aws-s3-v2[1.0.3].
```

I believe this is causing all of your build dependencies to not get installed. This would explain the failures in your `php artisan` commands, because Laravel itself may not be getting installed fully with everything you need.

Digging deeper in to the `aws/aws-sdk-php 2.8.31 requires guzzlehttp/guzzle ~3.7 -> no matching package found.` line above, I find that guzzle 3.x and 4.x are EOL (see readme in https://github.com/guzzle/guzzle). Is it possible that they are no longer available for download from the package repository?

Following the instructions from the [AWS PHP SDK documentation](http://docs.aws.amazon.com/aws-sdk-php/v2/guide/installation.html), I see the following output that shows it installing `guzzle/guzzle 3.9.3`  instead of `guzzlehttp/guzzle` which is what your build is trying to do. Is there something in your composer setup that is pinning guzzlehttp instead of guzzle? Is it possible that this is failing because the guzzlehttp repository doesn't have 3.x or 4.x releases as they are EOL? Here's the output from a container that I tried for comparison:

```
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 3 installs, 0 updates, 0 removals
  - Installing symfony/event-dispatcher (v2.8.26): Downloading (100%)
  - Installing guzzle/guzzle (v3.9.3): Downloading (100%)
  - Installing aws/aws-sdk-php (2.8.31): Downloading (100%)
symfony/event-dispatcher suggests installing symfony/dependency-injection ()
symfony/event-dispatcher suggests installing symfony/http-kernel ()
guzzle/guzzle suggests installing guzzlehttp/guzzle (Guzzle 5 has moved to a new package name. The package you have installed, Guzzle 3, is deprecated.)
aws/aws-sdk-php suggests installing doctrine/cache (Adds support for caching of credentials and responses)
aws/aws-sdk-php suggests installing ext-apc (Allows service description opcode caching, request and response caching, and credentials caching)
aws/aws-sdk-php suggests installing monolog/monolog (Adds support for logging HTTP requests and responses)
aws/aws-sdk-php suggests installing symfony/yaml (Eases the ability to write manifests for creating jobs in AWS Import/Export)
Package guzzle/guzzle is abandoned, you should avoid using it. Use guzzlehttp/guzzle instead.
Writing lock file
Generating autoload files
```

I believe this is what is blocking your builds right now. If we can get your build dependencies to install correctly your build should succeed. The build environment is exactly the same as the last successful build, so I cannot think of any other reason why it is failing.

**Comment by rageshkrishna** [comment]
It seems that this line is the key to the puzzle:

```
guzzle/guzzle suggests installing guzzlehttp/guzzle (Guzzle 5 has moved to a new package name. The package you have installed, Guzzle 3, is deprecated.)
```

guzzlehttp/guzzle only has 5.x releases and cannot satisfy a requirement for 3.x.

**Comment by aftabaslam** [comment]
@rageshkrishna 

Thanks for a detailed response, we did as suggested, i.e. replaced guzzlehttp with guzzle in our composer file and run the build, the results are no different. Please have a look at build results, it is still failing.

https://app.shippable.com/github/1eq/1EQ/runs/5244/1/console

**Comment by rageshkrishna** [comment]
There are still several unresolvable dependencies in the output of phing installBuildDeps. Most of them are reporting they require PHP 5.6 or above. Are you sure you did not change any other dependency versions? I am not sure how this could have succeeded previously on PHP 5.5 if the dependency itself is reporting an incompatibility.

Here is the first part of the output from your phing command that shows the problems:

````
Problem 1
    - Installation request for doctrine/annotations v1.4.0 -> satisfiable by doctrine/annotations[v1.4.0].
    - doctrine/annotations v1.4.0 requires php ^5.6 || ^7.0 -> your PHP version (5.5.9) does not satisfy that requirement.
  Problem 2
    - Installation request for doctrine/collections v1.4.0 -> satisfiable by doctrine/collections[v1.4.0].
    - doctrine/collections v1.4.0 requires php ^5.6 || ^7.0 -> your PHP version (5.5.9) does not satisfy that requirement.
  Problem 3
    - Installation request for doctrine/common v2.7.3 -> satisfiable by doctrine/common[v2.7.3].
    - doctrine/common v2.7.3 requires php ~5.6|~7.0 -> your PHP version (5.5.9) does not satisfy that requirement.
  Problem 4
    - Installation request for sly/notification-pusher v2.3.1 -> satisfiable by sly/notification-pusher[v2.3.1].
    - sly/notification-pusher v2.3.1 requires php >=5.6 -> your PHP version (5.5.9) does not satisfy that requirement.
  Problem 5
    - Installation request for zendframework/zend-validator 2.10.1 -> satisfiable by zendframework/zend-validator[2.10.1].
    - zendframework/zend-validator 2.10.1 requires php ^5.6 || ^7.0 -> your PHP version (5.5.9) does not satisfy that requirement.
  Problem 6
    - sly/notification-pusher v2.3.1 requires php >=5.6 -> your PHP version (5.5.9) does not satisfy that requirement.
    - davibennun/laravel-push-notification dev-laravel5 requires sly/notification-pusher 2.* -> satisfiable by sly/notification-pusher[v2.3.1].
    - Installation request for davibennun/laravel-push-notification dev-laravel5 -> satisfiable by davibennun/laravel-push-notification[dev-laravel5].
````

I also noticed that your YML is different now from what was tried in https://app.shippable.com/github/1eq/1EQ/runs/5236/1/console. The YML from that build will give you exactly the same build environment as the last succeeding build. Please use what that was doing to ensure the environment stays the same.

I am not sure what the resolution for the incompatible dependencies would be, though. It seems to me like an issue with your composer requirements.

**Comment by aftabaslam** [comment]
@rageshkrishna 
after updating the php version we are stuck on this plugin 

[PHP Error] include_once(Archive/Tar.php): failed to open stream: No such file or directory [line 75 of /root/.composer/vendor/phing/phing/classes/phing/tasks/ext/TarTask.php]
[PHP Error] include_once(): Failed opening 'Archive/Tar.php' for inclusion (include_path='/root/.composer/vendor/phing/phing/bin/../classes:/root/.composer/vendor/phing/phing/classes:.:') [line 75 of /root/.composer/vendor/phing/phing/classes/phing/tasks/ext/TarTask.php]

**Comment by rageshkrishna** [comment]
@aftabaslam https://github.com/websharks/phings/issues/87 tells me that you need to include the Archive_Tar dependency yourself in order to use TarTask. The linked issue talks about using `pear` to do it, but you are already using composer so maybe this is something that can be included in your composer.json? I'm afraid I only have a superficial knowledge of PHP, so I could be wrong about this...

On a related note, does this mean that the build issues have been isolated to problems with the dependencies themselves, or do you still feel that this is an issue with the build environment? I am curious about how you could get your steps to run locally on PHP 5.5 if the dependencies were reporting incompatibilities with that version.

**Comment by khurram85** [comment]
@rageshkrishna finally we got it working by updating .yml file to manually run composer install. I noticed regardless we install composer as part of our CI build even I am not sure if it's using the correct version of PHP. Have to use composer that is under `/root/.phpenv/shims/composer` to meet the dependencies. 

Regarding the Shippable upgrade process as a customer, we should receive notifications prior any new change going live to avoid surprises like this one. It took us almost 4 days to adapt to the changes Shippable made without notifying any stakeholders.
